homeassistant/components/light/hue.py

```diff
@@ -149,32 +150,53 @@ def update_lights():
             _LOGGER.exception("Cannot reach the bridge")
             return
 
-        api_states = api.get('lights')
+        api_lights = api.get('lights')
 
-        if not isinstance(api_states, dict):
+        if not isinstance(api_lights, dict):
+            _LOGGER.error("Got unexpected result from Hue API")
+            return
+
+        api_groups = api.get('groups')
+
+        if not isinstance(api_groups, dict):
             _LOGGER.error("Got unexpected result from Hue API")
             return
 
         new_lights = []
+        new_lightgroups = []
 
         api_name = api.get('config').get('name')
         if api_name in ('RaspBee-GW', 'deCONZ-GW'):
             bridge_type = 'deconz'
         else:
             bridge_type = 'hue'
 
-        for light_id, info in api_states.items():
+        for light_id, info in api_lights.items():
             if light_id not in lights:
                 lights[light_id] = HueLight(int(light_id), info,
                                             bridge, update_lights,
                                             bridge_type, allow_unreachable)
                 new_lights.append(lights[light_id])
             else:
                 lights[light_id].info = info
```

Comment:
Actually, this won't work either, since the id's of the lights and the light_groups aren't unique. I therefore need to store them separately (in `lights` and `lightgroups`). The need to merge both blocks to one will clutter this block with conditions to check is_group to know what list to check and where to store the items.